### DISCUSSION TITLE
Stuck on why status: 'pending', fetchStatus: 'idle' when using useReadContract

### DISCUSSION THREAD

**Discussion by smartydev42** [wevm/wagmi#3588] [Q&A]
Hey, 

I've got the following setup: 

```
////////////////////////////////////////////////
  // Wagmi Hook things
  const queryClient = useQueryClient();

  const { data, error, isLoading, isPending, isSuccess, queryKey, ...rest } =
    useReadContract({
      abi: abi,
      address: getContractAddress(),
      functionName: 'aFunctionName',
      query: {
        networkMode: 'online',
        enabled: true,
      },
    });
  // End Wagmi Hook things
  ////////////////////////////////////////////////
```

if i: console.log('Data is', data, rest); data is always undefined and data includes isPending = true status = pending and fetchStatus is idle. 

I am tearing my hair our and i'm really unsure what i've done wrong. 

I'm using avalanche if that helps, i've used wagmi before but likely about 6 months ago. so might be there's something that's changed?

**Comment by smartydev42** [comment]
ok, seems i found my issue, getContractAddress() returned nothing, was undefined. Seems i forgot to setup the .env. 

I would argue that useReadContract could probably do a "if address is null complain else do something"

**Comment by vseehausen** [comment]
Agree, it would be nice to get an error when necessary parameters are undefined. Like abi or address.
